src/cdi/CPEloss.hh

```diff
@@ -0,0 +1,141 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   cdi/CPEloss.hh
+ * \author Kendra Long
+ * \date   Fri Aug  2 14:28:08 2019
+ * \brief  CPEloss class header file (an abstract class)
+ * \note   Copyright (C) 2016-2019 Triad National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef __cdi_CPEloss_hh__
+#define __cdi_CPEloss_hh__
+
+#include "CPCommon.hh"
+#include "ds++/config.h"
+#include <string>
+#include <vector>
+
+namespace rtt_cdi {
+
+//========================================================================
+/*!
+ * \class CPEloss
+ *
+ * \brief This is a pure virtual class that defines a standard interface for
+ * all derived CPEloss objects.
+ *
+ * Any derived CPEloss object must provide as a minumum the functionality
+ * outlined in this routine.  This functionality includes access to the data
+ * grid and the ability to return calculated eloss values.
+ *
+ */
+//========================================================================
+
+class CPEloss {
+public:
+  // ---------- //
+  // Destructor //
+  // ---------- //
+
+  /*!
+   * \brief Default CPEloss() destructor.
+   *
+   * This is required to correctly release memory when any
+   * object derived from CPEloss is destroyed.
+   */
+  virtual ~CPEloss(){/*empty*/};
+
+  // --------- //
+  // Accessors //
+  // --------- //
+
+  /*!
+   * \brief Eloss accessor that returns a single eloss that 
+   *     corresponds to the provided temperature, density, and
+   *     incident particle speed.
+   *
+   * \param targetTemperature The temperature value for which an
+   *     eloss value is being requested (keV).
+   *
+   * \param targetDensity The density value for which an eloss 
+   *     value is being requested (g/cm^3).
+   *
+   * \param partSpeed The incident speed of the particle (cm/shk).
+   *
+   * \return A single Eloss rate coefficient (shk^-1).
+   */
+  virtual double getEloss(const double targetTemperature,
+                          const double targetDensity,
+                          const double partSpeed) const = 0;
+  /*!
+   * \brief Query whether the data is in tables or functional form.
+   */
+  virtual bool data_in_tabular_form() const = 0;
```

Comment:
Is there a use case for this function?

Reply:
Probably not? Not in our code, at least. I hemmed and hawed about several of the pure virtual functions, but ultimately left them in place because I'm not 100% how other codes might use these classes :)